sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/TransactionalBatchTest.java

```diff
@@ -0,0 +1,379 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.ISessionToken;
+import com.azure.cosmos.implementation.guava25.base.Function;
+import com.azure.cosmos.models.CosmosItemResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.assertj.core.data.Offset;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+import static com.azure.cosmos.implementation.batch.BatchRequestResponseConstant.MAX_OPERATIONS_IN_DIRECT_MODE_BATCH_REQUEST;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TransactionalBatchTest extends BatchTestBase {
+
+    private CosmosClient batchClient;
+    private CosmosContainer batchContainer;
+    private CosmosAsyncContainer batchAsyncContainer;
+
+    @Factory(dataProvider = "simpleClientBuildersWithDirect")
+    public TransactionalBatchTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void before_TransactionalBatchTest() {
+        assertThat(this.batchClient).isNull();
+        this.batchClient = getClientBuilder().buildClient();
+        batchAsyncContainer = getSharedMultiPartitionCosmosContainer(this.batchClient.asyncClient());
+        batchContainer = batchClient.getDatabase(batchAsyncContainer.getDatabase().getId()).getContainer(batchAsyncContainer.getId());
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        safeCloseSyncClient(this.batchClient);
+    }
+
+    @Test(groups = {"simple"}, timeOut = TIMEOUT)
+    public void batchExecutionRepeat() {
+        CosmosContainer container = this.batchContainer;
+        this.createJsonTestDocsAsync(container);
+
+        TestDoc firstDoc = this.populateTestDoc(this.partitionKey1);
+        TestDoc replaceDoc = this.getTestDocCopy(firstDoc);
+        replaceDoc.setCost(replaceDoc.getCost() + 1);
+
+        Mono<TransactionalBatchResponse> batchResponseMono = batchAsyncContainer.executeTransactionalBatch(
+            TransactionalBatch.createTransactionalBatch(this.getPartitionKey(this.partitionKey1))
+                .createItem(firstDoc)
+                .replaceItem(replaceDoc.getId(), replaceDoc));
```

Comment:
CosmosContainer provides sync API where as batchAsyncContainer provides async api.

you are mixing the async hierarchy with non-async hierarchy.

the usage expectation is once a user decide async vs non-async with stick to that decision everywhere and never mixes the usage of the two.

As these tests might get used by the end user as sample code we should ensure we are following above guideline.


please fix these in all tests. each test class should only stick to one single pattern and not mixes the two patterns

Reply:
All the other test sticks to sync container, as that verifies both sync and async. This test was to ensure that double block on Mono works as expected. But if users do see our test, I will move it to a separate class.